lib/task.js

```diff
@@ -221,11 +222,16 @@ export class Reclaimer {
     this.log('reclaiming task');
 
     try {
-      this.claim = await this.runtime.queue.reclaimTask(
-                                this.claim.status.taskId, this.claim.runId);
+      let queue = this.task.createQueue(this.claim.credentials);
+      this.claim = await queue.reclaimTask(
+                    this.claim.status.taskId, this.claim.runId);
       // reclaim does not return the task, so carry that forward from the previous
       // claim
       this.claim.task = task;
+
+      if (this.claim.status.taskId == this.primaryClaim.status.taskId) {
```

Comment:
I would have this replace line 292.  The only time we want to create it is when it's the primary task so that it can be used by the artifact handlers.